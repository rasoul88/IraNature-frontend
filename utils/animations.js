import { keyframes } from "styled-components";

export const moveInLeft = keyframes`
    0% {
        opacity: 0;
        transform: translateX(-10rem);
    }

    80% {
        transform: translateX(1rem);
    }

    100% {
        opacity: 1;
        transform: translateX(0);
    }
`;

export const moveInRight = keyframes`
    0% {
        opacity: 0;
        transform: translateX(10rem);
    }

    80% {
        transform: translateX(-1rem);
    }

    100% {
        opacity: 1;
        transform: translateX(0);
    }
`;

export const moveInBottom = keyframes`
    0% {
        opacity: 0;
        transform: translateY(3rem);
    }

    100% {
        opacity: 1;
        transform: translateY(0);
    }
`;

export const Shake = keyframes`
    0% {
        transform: translateX(-.3rem);
    }

    100% {
        transform: translateX(.3rem);
    }
`;

export const pageInRight = keyframes`
    0% {
        opacity: 0;
        transform: translateX(200%);
    }

    80% {
        transform: translateX(-10rem);
    }

    100% {
        opacity: 1;
        transform: translateX(0);
    }
`;
