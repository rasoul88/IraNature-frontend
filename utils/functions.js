import { toast } from "react-toastify";
import Spin from "../public/assets/icons/Spin.svg";

export const hexTorgba = (color, alpha) => {
  if (alpha === undefined) alpha = 1;
  const r = parseInt(color.substr(1, 2), 16);
  const g = parseInt(color.substr(3, 2), 16);
  const b = parseInt(color.substr(5, 2), 16);
  return `rgba(${r}, ${g}, ${b}, ${alpha})`;
};

export const createGradientbackground = (from, to) => {
  return `linear-gradient(to right bottom, ${from},${to})`;
};

export const createGradientText = (from, to) => {
  return `linear-gradient(to right bottom,${hexTorgba(from, 0.85)}, 
  ${hexTorgba(to, 0.85)})`;
};

export const showToast = (mode, message, onClose) => {
  toast.dismiss();
  toast[mode](message, {
    autoClose: 5000,
    theme: "colored",
    // onOpen: (props) => console.log("i opened"),
    onClose: onClose,
    style: {
      fontFamily: "inherit",
    },
    position: toast.POSITION.BOTTOM_CENTER,
  });
};

export const showLoadingToast = (message) => {
  toast.dismiss();
  toast.info(message, {
    autoClose: false,
    style: {
      fontFamily: "inherit",
      boxShadow: "0 1px 10px 0 rgb(0 0 0 / 30%), 0 2px 15px 0 rgb(0 0 0 / 25%)",
      color: "#333",
    },
    position: toast.POSITION.BOTTOM_CENTER,
    icon: <Spin style={{ transform: "scale(1.5)", fill: "#666" }} />,
  });
};

export const objectDateToString = (objDate) => {
  return `${objDate.year}/${objDate.month < 10 ? "0" : ""}${objDate.month}/${
    objDate.day < 10 ? "0" : ""
  }${objDate.day}`;
};

// export const stringDateToObject = (stringDate) => {
//   const arrDate = stringDate.split("/");
//   return { year: arrDate[0], month: arrDate[1], day: arrDate[2] };
// };
