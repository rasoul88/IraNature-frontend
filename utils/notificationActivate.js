import { post } from "axios";

const urlBase64ToUint8Array = (base64String) => {
  var padding = "=".repeat((4 - (base64String.length % 4)) % 4);
  var base64 = (base64String + padding).replace(/\-/g, "+").replace(/_/g, "/");

  var rawData = window.atob(base64);
  var outputArray = new Uint8Array(rawData.length);

  for (var i = 0; i < rawData.length; ++i) {
    outputArray[i] = rawData.charCodeAt(i);
  }
  return outputArray;
};

const configurePushSub = () => {
  if (!("serviceWorker" in navigator)) {
    return;
  }

  var reg;
  navigator.serviceWorker.ready
    .then(function (swreg) {
      reg = swreg;
      return swreg.pushManager.getSubscription();
    })
    .then(function (sub) {
      if (sub === null) {
        // Create a new subscription
        var vapidPublicKey =
          "BKMOXd1ARsXXz97HxAFb6tU2BmVdhDJkvpscv77UWadgtLidzKZUxHMloFHdOx18kuYxHkYUTHGzbW0I_cwwAyE";
        var convertedVapidPublicKey = urlBase64ToUint8Array(vapidPublicKey);
        return reg.pushManager.subscribe({
          userVisibleOnly: true,
          applicationServerKey: convertedVapidPublicKey,
        });
      } else {
      }
    })
    .then((newSub) => {
      if (newSub) return post("/subscriptions", newSub);
    })
    .then((res) => {
      console.log("new subscription", res);
    })
    .catch(function (err) {
      console.log(err);
    });
};

export const askForNotificationPermission = () => {
  Notification.requestPermission((result) => {
    if (result !== "granted") {
    } else {
      configurePushSub();
    }
  });
};
