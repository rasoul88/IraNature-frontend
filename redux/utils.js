import { objectDateToString } from "../utils/functions";
export const createQuery = (filterItems, dataLimits) => {
  let query = "";
  for (let item in filterItems) {
    if (item === "name" && filterItems[item].length > 0)
      query += `name[regex]=${filterItems[item]}&`;

    if (item === "days") {
      if (filterItems[item][0] > dataLimits[item].min)
        query += `duration[gte]=${filterItems[item][0]}&`;
      if (filterItems[item][1] < dataLimits[item].max)
        query += `duration[lte]=${filterItems[item][1]}&`;
    }

    if (item === "price") {
      if (filterItems[item][0] > dataLimits[item].min)
        query += `price[gte]=${filterItems[item][0]}&`;
      if (filterItems[item][1] < dataLimits[item].max)
        query += `price[lte]=${filterItems[item][1]}&`;
    }

    if (
      item === "dateRange" &&
      filterItems[item].from &&
      filterItems[item].to
    ) {
      query += `startDate[gte]=${objectDateToString(filterItems[item].from)}&`;
      query += `startDate[lte]=${objectDateToString(filterItems[item].to)}&`;
    }

    if (item === "startLocation" && filterItems[item] && filterItems[item][0]) {
      query += `startLocation=${filterItems[item][0].name}&`;
    }

    if (
      item === "difficulty" &&
      (filterItems[item].length === 1 || filterItems[item].length === 2)
    ) {
      filterItems[item].forEach((el) => (query += `difficulty=${el}&`));
    }

    if (
      item === "maxParticipants" &&
      filterItems[item][0] < dataLimits[item].max
    ) {
      query += `maxGroupSize[lte]=${filterItems[item][0]}&`;
    }

    if (item === "page") {
      query += `page=${filterItems[item]}&`;
    }
  }

  query = query.slice(0, query.length - 1);

  return query;
};
