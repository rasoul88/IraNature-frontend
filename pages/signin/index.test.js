import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { findByTestAttribute } from "../../test/utils";
import SignInSignUpPage from ".";

const setup = () => {
  return shallow(<SignInSignUpPage />);
};

test("renders signin-signup component", () => {
  const wrapper = setup();
  const component = findByTestAttribute(wrapper, "component-signin-signup");
  expect(component.length).toBe(1);
  expect(toJson(wrapper)).toMatchSnapshot();
});

describe("userReducer hook calls when click 'signup', 'signin', 'forgotPassword' and 'signinWithPassword' buttons", () => {
  let wrapper;
  const mockDispatch = jest.fn();
  beforeEach(() => {
    mockDispatch.mockClear();

    React.useReducer = jest.fn().mockReturnValue(["", mockDispatch]);

    wrapper = setup();
  });

  test("correct initial value ('sinin')", () => {
    expect(mockDispatch).not.toHaveBeenCalled();
  });

  test("click on 'signup' Button", () => {
    const signupButton = findByTestAttribute(wrapper, "signup-button");
    signupButton.simulate("click");
    expect(mockDispatch).toHaveBeenCalledWith({
      type: "setMode",
      payload: "signup",
    });
  });

  test("click on 'signin' Button", () => {
    const signinButton = findByTestAttribute(wrapper, "signin-button");
    signinButton.simulate("click");
    expect(mockDispatch).toHaveBeenCalledWith({
      type: "setMode",
      payload: "signin",
    });
  });

  describe("when state.forgotPassword is 'false'", () => {
    let forgotPasswordButton;
    let signinWithPasswordButton;

    beforeEach(() => {
      mockDispatch.mockClear();

      React.useReducer = jest.fn().mockReturnValue([
        {
          mode: "signin",
          forgotPassword: false,
        },
        mockDispatch,
      ]);

      wrapper = setup();

      forgotPasswordButton = findByTestAttribute(
        wrapper,
        "forgotPassword-button"
      );

      signinWithPasswordButton = findByTestAttribute(
        wrapper,
        "signinWithPassword-button"
      );
    });

    test("renders `forgotPassword` button correctly", () => {
      expect(forgotPasswordButton.length).toBe(1);
    });

    test("does not render `signinWithPassword` button", () => {
      expect(signinWithPasswordButton.length).toBe(0);
    });

    test("click on 'forgotPassword' Button", () => {
      forgotPasswordButton.simulate("click");
      expect(mockDispatch).toHaveBeenCalledWith({
        type: "forgotPassword",
        payload: true,
      });
    });
  });

  describe("when state.forgotPassword is 'true'", () => {
    let forgotPasswordButton;
    let signinWithPasswordButton;

    beforeEach(() => {
      mockDispatch.mockClear();

      React.useReducer = jest.fn().mockReturnValue([
        {
          mode: "signin",
          forgotPassword: true,
        },
        mockDispatch,
      ]);

      wrapper = setup();

      forgotPasswordButton = findByTestAttribute(
        wrapper,
        "forgotPassword-button"
      );

      signinWithPasswordButton = findByTestAttribute(
        wrapper,
        "singinWithPassword-button"
      );
    });

    test("renders `forgotPassword` button correctly", () => {
      expect(forgotPasswordButton.length).toBe(0);
    });

    test("does not render `signinWithPassword` button", () => {
      expect(signinWithPasswordButton.length).toBe(1);
    });

    test("click on 'forgotPassword' Button", () => {
      signinWithPasswordButton.simulate("click");
      expect(mockDispatch).toHaveBeenCalledWith({
        type: "forgotPassword",
        payload: false,
      });
    });
  });
});
