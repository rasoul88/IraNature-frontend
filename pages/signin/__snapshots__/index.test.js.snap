// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders signin-signup component 1`] = `
<indexstyles__SectionContainer
  data-test="component-signin-signup"
  id="signin-signup"
  mode="signin"
>
  <indexstyles__FormsContainer
    data-aos="fade-up"
    data-aos-anchor-placement="top-bottom"
    data-aos-delay="500"
    data-aos-duration="1000"
  >
    <indexstyles__SignInSignUp
      mode="signin"
    >
      <indexstyles__SignInForm
        action=""
        mode="signin"
        onSubmit={[Function]}
      >
        <indexstyles__SignInFormContainer>
          <indexstyles__TitleContainer>
             ورود
          </indexstyles__TitleContainer>
          <indexstyles__InputContainer>
            <indexstyles__InputField
              placeholder="نام کاربری"
              required={true}
              type="text"
            />
            <test-file-stub />
          </indexstyles__InputContainer>
          <indexstyles__InputContainer>
            <indexstyles__InputField
              placeholder="کلمه عبور"
              required={true}
              type="password"
            />
            <test-file-stub />
          </indexstyles__InputContainer>
          <indexstyles__SubmitButton
            solid={true}
            type="submit"
            value="ورود به حساب "
          />
          <indexstyles__ForgotPasswordButton
            data-test="forgotPassword-button"
            onClick={[Function]}
          >
            رمز عبور خود را فراموش کرده اید؟
          </indexstyles__ForgotPasswordButton>
          <indexstyles__SocialTextContainer>
            با شبکه های اجتماعی زیر وارد شوید
          </indexstyles__SocialTextContainer>
          <indexstyles__SocialMediaContainer>
            <indexstyles__SocialIconContainer
              color="#8a3ab9"
            >
              <test-file-stub />
            </indexstyles__SocialIconContainer>
            <indexstyles__SocialIconContainer
              color="#00acee"
            >
              <test-file-stub />
            </indexstyles__SocialIconContainer>
            <indexstyles__SocialIconContainer
              color="#ea4335"
            >
              <test-file-stub />
            </indexstyles__SocialIconContainer>
            <indexstyles__SocialIconContainer
              color="#0e76a8"
            >
              <test-file-stub />
            </indexstyles__SocialIconContainer>
          </indexstyles__SocialMediaContainer>
        </indexstyles__SignInFormContainer>
      </indexstyles__SignInForm>
      <indexstyles__SignUpForm
        action=""
        mode="signin"
      >
        <indexstyles__TitleContainer>
          ثبت نام
        </indexstyles__TitleContainer>
        <indexstyles__InputContainer>
          <indexstyles__InputField
            placeholder="نام کاربری"
            required={true}
            type="text"
          />
          <test-file-stub />
        </indexstyles__InputContainer>
        <indexstyles__InputContainer>
          <indexstyles__InputField
            placeholder="ایمیل"
            required={true}
            type="email"
          />
          <test-file-stub />
        </indexstyles__InputContainer>
        <indexstyles__InputContainer>
          <indexstyles__InputField
            placeholder="کلمه عبور"
            required={true}
            type="password"
          />
          <test-file-stub />
        </indexstyles__InputContainer>
        <indexstyles__SubmitButton
          type="submit"
          value="ثبت نام"
        />
        <indexstyles__SocialTextContainer>
          با شبکه های اجتماعی زیر ثبت نام کنید
           
        </indexstyles__SocialTextContainer>
        <indexstyles__SocialMediaContainer>
          <indexstyles__SocialIconContainer
            color="#8a3ab9"
          >
            <test-file-stub />
          </indexstyles__SocialIconContainer>
          <indexstyles__SocialIconContainer
            color="#00acee"
          >
            <test-file-stub />
          </indexstyles__SocialIconContainer>
          <indexstyles__SocialIconContainer
            color="#ea4335"
          >
            <test-file-stub />
          </indexstyles__SocialIconContainer>
          <indexstyles__SocialIconContainer
            color="#0e76a8"
          >
            <test-file-stub />
          </indexstyles__SocialIconContainer>
        </indexstyles__SocialMediaContainer>
      </indexstyles__SignUpForm>
    </indexstyles__SignInSignUp>
  </indexstyles__FormsContainer>
  <indexstyles__PannelsContainer>
    <indexstyles__LeftPannelsContainer
      mode="signin"
    >
      <indexstyles__ContentContainer>
        <h3>
          حساب کاربری ندارید؟
        </h3>
        <p>
           
          لطفا بر روی دکمه ی زیر کلیک کنید و با ایجاد حساب کاربری از تمام مزایای وب سایت ما بهره مند شوید
        </p>
        <button
          data-test="signup-button"
          onClick={[Function]}
        >
          ثبت نام
        </button>
      </indexstyles__ContentContainer>
      <indexstyles__ImageContainer
        alt="login"
        src="/assets/icons/register.svg"
      />
    </indexstyles__LeftPannelsContainer>
    <indexstyles__RightPannelsContainer
      mode="signin"
    >
      <indexstyles__ContentContainer>
        <h3>
          حساب کاربری دارید؟
        </h3>
        <p>
           
          لطفا بر روی دکمه ی زیر کلیک کنید و با ورود به حساب کاربری که قبلا ایجاد کرده اید از تمام مزایای وب سایت ما بهره مند شوید
        </p>
        <button
          data-test="signin-button"
          onClick={[Function]}
        >
          ورود
        </button>
      </indexstyles__ContentContainer>
      <indexstyles__ImageContainer
        alt="register"
        src="/assets/icons/Login2.svg"
      />
    </indexstyles__RightPannelsContainer>
  </indexstyles__PannelsContainer>
</indexstyles__SectionContainer>
`;
